diff --git a/cgltf.h b/cgltf.h
index 316a11d..40823e5 100644
--- a/cgltf.h
+++ b/cgltf.h
@@ -2219,11 +2219,23 @@ static cgltf_ssize cgltf_component_read_integer(const void* in, cgltf_component_
 	switch (component_type)
 	{
 		case cgltf_component_type_r_16:
-			return *((const int16_t*) in);
+		{
+			int16_t v;
+			memcpy(&v, in, sizeof(v));
+			return v;
+		}
 		case cgltf_component_type_r_16u:
-			return *((const uint16_t*) in);
+		{
+			uint16_t v;
+			memcpy(&v, in, sizeof(v));
+			return v;
+		}
 		case cgltf_component_type_r_32u:
-			return *((const uint32_t*) in);
+		{
+			uint32_t v;
+			memcpy(&v, in, sizeof(v));
+			return v;
+		}
 		case cgltf_component_type_r_8:
 			return *((const int8_t*) in);
 		case cgltf_component_type_r_8u:
@@ -2238,9 +2250,17 @@ static cgltf_size cgltf_component_read_index(const void* in, cgltf_component_typ
 	switch (component_type)
 	{
 		case cgltf_component_type_r_16u:
-			return *((const uint16_t*) in);
+		{
+			uint16_t v;
+			memcpy(&v, in, sizeof(v));
+			return v;
+		}
 		case cgltf_component_type_r_32u:
-			return *((const uint32_t*) in);
+		{
+			uint32_t v;
+			memcpy(&v, in, sizeof(v));
+			return v;
+		}
 		case cgltf_component_type_r_8u:
 			return *((const uint8_t*) in);
 		default:
@@ -2252,7 +2272,9 @@ static cgltf_float cgltf_component_read_float(const void* in, cgltf_component_ty
 {
 	if (component_type == cgltf_component_type_r_32f)
 	{
-		return *((const float*) in);
+		float v;
+		memcpy(&v, in, sizeof(v));
+		return v;
 	}
 
 	if (normalized)
@@ -2261,9 +2283,17 @@ static cgltf_float cgltf_component_read_float(const void* in, cgltf_component_ty
 		{
 			// note: glTF spec doesn't currently define normalized conversions for 32-bit integers
 			case cgltf_component_type_r_16:
-				return *((const int16_t*) in) / (cgltf_float)32767;
+			{
+				int16_t v;
+				memcpy(&v, in, sizeof(v));
+				return v / (cgltf_float)32767;
+			}
 			case cgltf_component_type_r_16u:
-				return *((const uint16_t*) in) / (cgltf_float)65535;
+			{
+				uint16_t v;
+				memcpy(&v, in, sizeof(v));
+				return v / (cgltf_float)65535;
+			}
 			case cgltf_component_type_r_8:
 				return *((const int8_t*) in) / (cgltf_float)127;
 			case cgltf_component_type_r_8u:
